### PR TITLE
479 mirai pr

### PR DESCRIPTION
**Pull request by pawelru**
most of my code review comments

please note that the target is a feature branch

### CODE REVIEW

File: R/init.R [hunk 1]
```diff
@@ -154,7 +154,13 @@ init <- function(data,
...
 
   # invoke lockfile creation
-  process <- teal_lockfile_invoke()
+  user_lockfile <- getOption("teal.renv.lockfile", "")
+  lockfile_process <- NULL
+  if (!identical(user_lockfile, "")) {
+    teal_lockfile_external()
+  } else {
+    lockfile_process <- teal_lockfile_process_invoke()
+  }
```
Comment (m7pr) [inline R/init.R:163]: This could also be wrapped into a high-level function

File: R/teal_lockfile.R [hunk 2]
```diff
@@ -122,47 +128,46 @@ renv_snapshot <- function(lockfile_path = NULL, opts, sysenv, libpaths, wd) {
...
-      " with error(s)"
-    } else {
-      ""
-    }
+  observeEvent(process$status(), {
+    if (process$status() == "initial" || process$status() == "running") {
+      shinyjs::html("teal-lockFileStatus", "Creating lockfile...")
+    } else if (process$status() == "success") {
+      result <- process$result()
+      browser()
```
Comment (m7pr) [inline R/teal_lockfile.R:146]: @pawelru :P
```suggestion

```

**Comment by m7pr** [comment]
Looks good. Love the way a place for the download link gets updated with an informative status

File: R/init.R (hunk 1 above)
Comment (pawelru) [reply R/init.R]: added: 2a931b52c9ffec9ba92f29f05c76ab48619828e1

**Review by m7pr — APPROVED** [review]
Hey @pawelru let's merge this to the original mirai-lockfile PR and let's add shinytest2 tests in there

**Comment by pawelru** [comment]
ok thanks
It would be great to also add this https://github.com/insightsengineering/teal/issues/1276

**Pull request merged**